virtwho/executor.py

```diff
@@ -68,8 +74,36 @@ def _create_virt_backends(self):
                 self.logger.error('Unable to use configuration "%s": %s', name, str(e))
                 continue
             virts.append(virt)
+        self._init_run_status(config_names)
         return virts
 
+    def _init_run_status(self, config_names=[]):
+        # ensure that there is an entry in the status file for each config. We don't care if they work, we
+        # need to record their existence
+        lock = PIDLock(STATUS_LOCK)
+        while lock.is_locked():
+            time.sleep(1)
+        with lock:
+            os.makedirs(STATUS_DATA_DIR, exist_ok=True)
+            # need to create the file if it does not exist
+            with open(STATUS_DATA, "w+") as json_status:
```

Comment:
We read the JSON file here. Thus the file should be opened using in read ("r") mode not "w+".

Reply:
w+ plus is used so that the file is created when it does not exist.
I added to the comment.